Treat MYPAINT_BRUSHES_DIR as a value that has not already been quoted.

See: https://github.com/mypaint/mypaint-brushes/issues/5
--- app/config/gimpcoreconfig.c.orig	2020-06-02 22:16:25.000000000 -0500
+++ app/config/gimpcoreconfig.c	2020-06-02 22:16:26.000000000 -0500
@@ -51,6 +51,9 @@
 #include "gimp-intl.h"
 
 
+#define _QUOTE(x) #x
+#define QUOTE(x) _QUOTE(x)
+
 #define GIMP_DEFAULT_BRUSH         "2. Hardness 050"
 #define GIMP_DEFAULT_DYNAMICS      "Dynamics Off"
 #define GIMP_DEFAULT_PATTERN       "Pine"
@@ -314,7 +317,7 @@
                                       "share", "mypaint-data",
                                       "1.0", "brushes", NULL);
 #else
-  mypaint_brushes = g_strdup (MYPAINT_BRUSHES_DIR);
+  mypaint_brushes = g_strdup (QUOTE(MYPAINT_BRUSHES_DIR));
 #endif
 
   path = g_build_path (G_SEARCHPATH_SEPARATOR_S,
--- app/core/gimpdata.c.orig	2020-02-23 12:21:08.000000000 -0600
+++ app/core/gimpdata.c	2020-06-02 21:38:28.000000000 -0500
@@ -36,6 +36,9 @@
 #include "gimp-intl.h"
 
 
+#define _QUOTE(x) #x
+#define QUOTE(x) _QUOTE(x)
+
 enum
 {
   DIRTY,
@@ -457,10 +460,10 @@
           identifier = g_filename_to_utf8 (tmp, -1, NULL, NULL, NULL);
           g_free (tmp);
         }
-      else if (g_str_has_prefix (path, MYPAINT_BRUSHES_DIR))
+      else if (g_str_has_prefix (path, QUOTE(MYPAINT_BRUSHES_DIR)))
         {
           tmp = g_strconcat ("${mypaint_brushes_dir}",
-                             path + strlen (MYPAINT_BRUSHES_DIR),
+                             path + strlen (QUOTE(MYPAINT_BRUSHES_DIR)),
                              NULL);
           identifier = g_filename_to_utf8 (tmp, -1, NULL, NULL, NULL);
           g_free (tmp);
